nibabel/filebasedimages.py

```diff
@@ -47,19 +52,19 @@ def from_header(klass, header=None):
         )
 
     @classmethod
-    def from_fileobj(klass, fileobj: io.IOBase):
-        raise NotImplementedError
+    def from_fileobj(klass: type[HdrT], fileobj: io.IOBase) -> HdrT:
+        raise NotImplementedError  # pragma: no cover
 
-    def write_to(self, fileobj: io.IOBase):
-        raise NotImplementedError
+    def write_to(self, fileobj: io.IOBase) -> None:
+        raise NotImplementedError  # pragma: no cover
 
-    def __eq__(self, other):
-        raise NotImplementedError
+    def __eq__(self, other: object) -> bool:
```

Comment:
`ty.Any` would mean that the internals do not get type checked and you might raise an `AttributeError` or something else that will break rather than return True/False. There's a good StackOverflow thread: https://stackoverflow.com/questions/57905233/how-to-annotate-eq-when-it-uses-an-attribute